File: Factorial-100/Factorial100.pas
program Factorial100;

var
  i      : longint;
  value  : integer;
  result : int64;

function factorial( value : integer ) : int64;
var
  result    : int64;
  iteration : integer;
begin
  result := 1;
  for iteration := 1 to value do
  begin
    result := result * iteration;
  end;

  factorial := result;
end;

begin
  for i := 0 to 50000 do
  begin
    for value := 0 to 100 do
    begin
      result := factorial( value );
      WriteLn( result );
    end;
  end;
end.

File: Factorial-100/Factorial100.go
package main

import "fmt"

func factorial( value int ) int64 {
  var result int64 = 1

  for iteration := 1; iteration <= value; iteration++ {
    result = result * int64( iteration )
  }

  return result
}

func main() {
  var result int64

  for value := 0; value <= 50000; value++ {
    for value := 0; value <= 100; value++ {
      result = factorial( value )
      fmt.Println( result )
    }
  }
}
